electricitylci/natural_gas_upstream.py

```diff
@@ -0,0 +1,142 @@
+    #the fuel consumption for generating electricity for each facility
+    #and fuel type. Filter the data to only include NG facilities and on positive
+    #fuel consumption. Group that data by Plant Id as it is possible to have
+    #multiple rows for the same facility and fuel based on different prime
+    #movers (e.g., gas turbine and combined cycle). 
+    eia_generation_data = eia923_download_extract(eia_gen_year)
+    
+    column_filt = ((eia_generation_data['Reported Fuel Type Code'] == 'NG') & 
+                   (eia_generation_data['Total Fuel Consumption MMBtu'] > 0))
+    ng_generation_data = eia_generation_data[column_filt]
+    
+    ng_generation_data = ng_generation_data.groupby('Plant Id').agg(
+            {'Total Fuel Consumption MMBtu':'sum'}).reset_index()
+    ng_generation_data['Plant Id'] = ng_generation_data['Plant Id'].astype(int)
+    
+    
+    #Import the mapping file which has the source gas basin for each Plant Id. 
+    #Merge with ng_generation dataframe.
+    
+    ng_basin_mapping = pd.read_csv(data_dir + 'gas_supply_basin_mapping.csv')
+    
+    subset_cols = ['Plant Code', 'NG_LCI_Name']
+    
+    ng_basin_mapping = ng_basin_mapping[subset_cols]
+    
+    ng_generation_data_basin = pd.merge(
+            left = ng_generation_data, 
+            right = ng_basin_mapping, 
+            left_on = 'Plant Id', 
+            right_on = 'Plant Code')
+    
+    ng_generation_data_basin = ng_generation_data_basin.drop(
+            columns = ['Plant Code'])
+    
+    #Read the NG LCI excel file
+    ng_lci = pd.read_excel(
+            data_dir + 'NG_LCI.xlsx', 
+            sheetname = 'Basin_Mean_Data')
+    
+    ng_lci_air = ng_lci[ng_lci['Compartment'] == 'Air']
+    
+    #Getting list of column values which are basically the emissions names
+    col_list = list(ng_lci.columns.get_values())
+    emissions_list = col_list[3:]
+    
+    # This line can go away once the elementary flow names are replaced in the 
+    #NG_LCI.xlsx file.
+    emissions_list = [emission.strip(' (kg/MJ)') for emission in emissions_list]
+    
+    
+    stage_list = list(ng_lci['Stage'].unique())
+    
+    ng_lci_cols = ['Basin', 'Compartment', 'Stage'] + emissions_list
+    
+    ng_lci.columns = ng_lci_cols
+    
+    #Merge basin data with LCI dataset
+    ng_lci_basin = pd.merge(
+            ng_lci, 
+            ng_generation_data_basin, 
+            left_on = 'Basin', 
+            right_on = 'NG_LCI_Name')
+    
+    
+    #Multiplying with the EIA 923 fuel consumption; conversion factor is 
+    #for MMBtu to MJ
+    for i in emissions_list:
+        ng_lci_basin[i] = (
+                ng_lci_basin[i] * 
+                ng_lci_basin['Elec Fuel Consumption MMBtu'] * 
+                1055.056)
+        
+    ng_lci_basin = ng_lci_basin.drop(
+            ['NG_LCI_Name', 'Elec Fuel Consumption MMBtu'], axis = 1)
+    
+    #Output is kg emission for 2016 by facility Id, not normalized to 
```

Comment:
Remove year from comment since it'll change.

Reply:
Done.